android/src/main/java/io/mosip/tuvali/openid4vpble/Openid4vpBleModule.kt

```diff
@@ -119,13 +120,14 @@ class Openid4vpBleModule(private val reactContext: ReactApplicationContext) :
   fun destroyConnection(callback: Callback) {
     //TODO: Make sure callback can be called only once[Can be done once wallet and verifier split into different modules]
     Log.d(logTag, "destroyConnection called at ${System.nanoTime()}")
-
-    stopBLE(callback)
+    safeExecute.run {
+      stopBLE(callback)
+    }
   }
 
   private fun stopBLE(callback: Callback) {
-    synchronized(mutex) {
-      if (wallet == null && verifier == null) {
+    safeExecute.run {
```

Comment:
Looks like there is cycling call stack problem here

stopBLE -> safeExecute.run -> handleException -> stopBLE